in_toto/runlib.py

```diff
@@ -818,11 +843,21 @@ def in_toto_record_stop(step_name, product_list, signing_key=None,
     unfinished_fn = UNFINISHED_FILENAME_FORMAT.format(step_name=step_name,
         keyid=signing_key["keyid"])
 
+    if metadata_directory is not None:
+      unfinished_fn = os.path.join(metadata_directory, unfinished_fn)
+
   # FIXME: Currently there is no way to know the default GPG key's keyid and
   # so we glob for preliminary link files
+  # FIXME: UNFINISHED_FILENAME_FORMAT_GLOB_DIR has a problem with window system
   else:
     unfinished_fn_glob = UNFINISHED_FILENAME_FORMAT_GLOB.format(
```

Comment:
I think it may be nicer to wrap this as part of the `if metadata_directory` condition below, instead of executing both in case a value is passed. It also slightly improves readability IMO.

This is assuming we stick with `UNFINISHED_FILENAME_FORMAT_GLOB_DIR` of course.